### PR TITLE
Make header guards uniform

### PR DESCRIPTION
**Pull request by peterschrammel**
Following the scheme CPROVER_DIRECTORIES_FILE_H

### CODE REVIEW

File: src/aa-symex/path_search.h [hunk 1]
```diff
@@ -68,7 +68,7 @@ class path_searcht:public safety_checkert
 
   // returns whether at least one child process has terminated
   bool try_await();
-#endif
+#endif // CPROVER_AA_SYMEX_PATH_SEARCH_H
```
Comment (tautschnig) [inline src/aa-symex/path_search.h]: This comment is misplaced (it's not the #endif at the end of the file)

File: src/analyses/goto_rw.h [hunk 2]
```diff
@@ -85,7 +85,7 @@ class rw_range_sett
   typedef std::map<irep_idt, range_domain_baset*> objectst;
   #else
   typedef hash_map_cont<irep_idt, range_domain_baset*, string_hash> objectst;
-  #endif
+  #endif // CPROVER_ANALYSES_GOTO_RW_H
```
Comment (tautschnig) [inline src/analyses/goto_rw.h]: Misplaced comment

File: src/analyses/natural_loops.h [hunk 3]
```diff
@@ -84,7 +84,7 @@ Function: natural_loops_templatet::compute
 
 #ifdef DEBUG
 #include <iostream>
-#endif
+#endif // CPROVER_ANALYSES_NATURAL_LOOPS_H
```
Comment (tautschnig) [inline src/analyses/natural_loops.h]: Misplaced comment

File: src/analyses/natural_loops.h [hunk 4]
```diff
@@ -93,7 +93,7 @@ void natural_loops_templatet<P, T>::compute(P &program)
 
 #ifdef DEBUG
   cfg_dominators.output(std::cout);
-#endif
+#endif // CPROVER_ANALYSES_NATURAL_LOOPS_H
```
Comment (tautschnig) [inline src/analyses/natural_loops.h]: Misplaced comment

File: src/analyses/natural_loops.h [hunk 5]
```diff
@@ -114,7 +114,7 @@ void natural_loops_templatet<P, T>::compute(P &program)
           std::cout << "Computing loop for "
                     << m_it->location_number << " -> "
                     << (*n_it)->location_number << "\n";
-#endif
+#endif // CPROVER_ANALYSES_NATURAL_LOOPS_H
```
Comment (tautschnig) [inline src/analyses/natural_loops.h]: Misplaced comment

File: src/ansi-c/library/jsa.h [hunk 6]
```diff
@@ -1,10 +1,10 @@
 /* FUNCTION: __CPROVER_jsa_synthesise */
 
-#ifndef JSA_TRANSFORMERS_H_
+#ifndef CPROVER_ANSI_C_LIBRARY_JSA_H
 
 #ifdef JSA_GENETIC_SYNTHESIS_H_
 #ifndef __CPROVER_jsa_extern
-#define __CPROVER_jsa_extern extern
+#define CPROVER_ANSI_C_LIBRARY_JSA_H
```
Comment (tautschnig) [inline src/ansi-c/library/jsa.h]: This #define must not be rewritten

File: src/java_bytecode/bytecode_info.h [hunk 7]
```diff
@@ -1,3 +1,6 @@
+#ifndef CPROVER_JAVA_BYTECODE_BYTECODE_INFO_H
+#define CPROVER_JAVA_BYTECODE_BYTECODE_INFO_H
+
 #pragma once
```
Comment (tautschnig) [inline src/java_bytecode/bytecode_info.h]: We should get rid of `#pragma once` (which is exactly why we are using header guards)

**Comment by peterschrammel** [comment]
@tautschnig, updated.

**Pull request merged**